NOTES.md

```diff
@@ -1,3 +1,186 @@
+* Improved support for asynchronous queries
+* Added a new feature called `tables` which can be used to store JSON documents
+
+Read the [blog post][2.0-RC1-blog] for more details.
+
+[2.0-RC1-blog]: http://rethinkdb.com/blog/2.0-release-candidate/
+
+## Compatibility ##
+
+This release may not be compatible with RethinkDB 2.0.
+
+### Handling of negative zero in indexes ###
+
+IEEE 754 floating point numbers distinguish between negative (-0) and
+positive (+0) zero.  The following information is only relevant if you
+are storing negative zero values in your documents.  We expect very
+few users to be affected by this change.
+
+ReQL compares -0 and +0 as equal in accordance with IEEE 754.  In
+previous versions of RethinkDB, -0 and +0 were however treated as
+distinct values in primary and secondary indexes.  This could lead to
+inconsistent behavior and wrong query results in some rare cases.
+
+Starting with RethinkDB 2.0, -0 and +0 are indexed as equal values.
+Secondary indexes can be [migrated online][1.14-outdated-index] using
+the `rethinkdb index-rebuild` utility.  If any of your documents have
+negative zero values in their primary keys, those documents will
+become partially inaccessible in RethinkDB 2.0. You will need to
+re-import the affected tables using the `rethinkdb dump` and
+`rethinkdb restore` commands. See [Back up your data][backup-docs] for
+more information.
+
+If you are unsure if any of your documents are affected, you can run
+`python -m rethinkdb._negative_zero_check` after upgrading both the
+server and Python driver.  See the output of `python -m
+rethinkdb._negative_zero_check --help` for additional options.
+
+[1.14-outdated-index]: http://rethinkdb.com/docs/troubleshooting/#my-secondary-index-is-outdated
+[backup-docs]: http://www.rethinkdb.com/docs/backup/
+
+### API-breaking changes ###
+
+* `between` no longer accepts `null` bounds. The new `r.minval` and
+  `r.maxval` can be used instead
+
+* The `any` and `all` commands have been removed. The `or` and `and`
+  commands can be used instead
+
+* `indexes_of` has been renamed to `offsets_of`
+
+* The `squash` argument to `changes` now defaults to `false`
+
+* The type hierarchy for exception types in the Python driver changed.
+  All exceptions including `RqlDriverError` now inherit from the
+  `RqlError` type.  `RqlRuntimeError`, `RqlCompileError` and
+  `RqlClientError` additionally inherit from the new `RqlQueryError`
+  type.
+
+## New Features ##
+
+* ReQL
+  * Added support for changefeeds on `get_all` and `union` queries (#3642)
+  * `between` no longer accepts `null` as a bound. The new `r.minval` and `r.maxval` can be used instead (#1023)
+  * Added support for getting the state of a changefeed using the new `include_states` optarg to `changes` (#3709)
+* Drivers
+  * Added support for non-blocking `cursor.get` (#3529)
+  * Added support for parallel query execution on a single connection (#3754)
```

Comment:
To distinguish from the parallel execution of different parts of a single query, we should maybe rephrase this:
"Added support for executing multiple queries in parallel on a single connection"